### PR TITLE
Product card update

### PR DESCRIPTION
**Pull request by Gray754**

### CODE REVIEW

File: src/ProductCard/StoreLocationButton.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React, { FunctionComponent } from 'react'
```
Comment (craig1123) [inline src/ProductCard/StoreLocationButton.tsx:1]: This should really be on the mattress matcher project. No other project will consume this, it's too specific

File: src/ProductCard/StoreLocationButton.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import styles from './productCard.module.scss'
+
+interface Props {
+  storeLocation?: string
+  onClick?: Function
+}
+
+const StoreLocationButton: FunctionComponent<Props> = ({ storeLocation, onClick }) => {
+  const handleClick = () => {
```
Comment (craig1123) [inline src/ProductCard/StoreLocationButton.tsx:14]: this function shouldn't exist. Just pass `onClick` to the onClick on the button comp. 

```
<button onClick={onClick}>
```

Comment (craig1123) [reply src/ProductCard/StoreLocationButton.tsx:14]: if the the prop onClick is undefined, then the event listener won't be passed the Element

File: src/ProductCard/StoreLocationButton.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      onClick
+    }
+  }
+
+  const marker = storeLocation && <MapMarker fillColor="#d63426" className={styles.mapMarker} />
+  const arrow = storeLocation && <ChevronRight style={{ transform: 'rotate(90deg)', marginRight: '10px' }} />
+  const btnText = storeLocation ? <span className={styles.mobileColor}>Available to try in-store</span> : 'Online Only'
+
+  return (
+    <button className={clsx(styles['store-location-btn'], storeLocation && styles.align)} onClick={() => handleClick()}>
```
Comment (craig1123) [inline src/ProductCard/StoreLocationButton.tsx:25]: Button needs a "type". If you don't specify a type, it's "submit" by default. You don't want that unless this is in a form. The default behavior is to refresh the page with type="submit" unless you do an event.preventDefault() which I see you do not have

File: src/ProductCard/StoreLocationButton.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+const StoreLocationButton: FunctionComponent<Props> = ({ storeLocation, onClick }) => {
+  const handleClick = () => {
+    if (onClick) {
+      onClick
+    }
+  }
+
+  const marker = storeLocation && <MapMarker fillColor="#d63426" className={styles.mapMarker} />
+  const arrow = storeLocation && <ChevronRight style={{ transform: 'rotate(90deg)', marginRight: '10px' }} />
```
Comment (craig1123) [inline src/ProductCard/StoreLocationButton.tsx:21]: Isn't there a ChevronLeft component that already does this?

File: src/ProductCard/index.tsx [hunk 5]
```diff
@@ -25,6 +27,11 @@ interface Props {
   matchPercentage?: number
   /** discounted price or range of discounted prices if any */
   discountPrice?: number[]
+  storeLocation?: string
+  productPage?: string
+  deliveryDate?: string
+  financeLink?: string
+  storeLocationBtnOnClick?: Function
```
Comment (craig1123) [inline src/ProductCard/index.tsx:34]: What are the arguments of the Function? What is being returned?
Better to do this: 
```
() => void
```
You can put the arguments and their types as well as what the return type will be

File: src/ProductCard/index.tsx [hunk 6]
```diff
@@ -55,22 +67,48 @@ const ProductCard: FunctionComponent<Props> = ({
 
   const ratings = rating && (
     <div className={styles['rating-row']}>
-      <Rating data-testid="star-rating" name="product-rating" value={rating} readOnly size="sm" />
+      <Rating
+        data-testid="star-rating"
```
Comment (craig1123) [inline src/ProductCard/index.tsx:71]: is this needed? Can't query this Element?

File: src/ProductCard/index.tsx [hunk 7]
```diff
@@ -55,22 +67,48 @@ const ProductCard: FunctionComponent<Props> = ({
...
+        size="sm"
+      />
       <span className={styles.reviews}>
         <small className={styles['average-rating']}>{rating}/5</small>{' '}
         <small data-testid="review-count">({reviews})</small>
       </span>
     </div>
   )
 
+  const deliveryDetails = (
```
Comment (craig1123) [inline src/ProductCard/index.tsx:85]: Too specific for this to be in the component library; remove

File: src/ProductCard/index.tsx [hunk 8]
```diff
@@ -55,22 +67,48 @@ const ProductCard: FunctionComponent<Props> = ({
...
+          Delivery as soon as: <span className={styles['delivery-date']}>{deliveryDate}</span>
+        </p>
+      )}
+    </div>
+  )
+
   const priceInfo = (
     <div className={styles['price-container']}>
+      <span className={styles.mobileBtnContainer}>
+        <StoreLocationButton storeLocation={storeLocation} onClick={storeLocationBtnOnClick} />
```
Comment (craig1123) [inline src/ProductCard/index.tsx:103]: You can make a prop for this to be passed down. It IS possible to pass down react components like Children. You could even do render props. Regardless, I wouldn't include this in the CL

File: src/ProductCard/StoreLocationButton.tsx (hunk 1 above)
Comment (Gray754) [reply src/ProductCard/StoreLocationButton.tsx:1]: The problem with having this on the MM project is that the specific location for it in the designs. If it was just a button below the other pieces it works, but on the mobile design the button turns smaller and goes next to the price.

**Comment by craig1123** [comment]
@Gray754 Honestly, I would start out in MM results page before adding it here

**Pull request closed without merging**